### PR TITLE
Add DestroyContext command.

### PR DESCRIPTION
**Pull request by zhalvorsen**

### CODE REVIEW

File: dpe/src/dpe_instance.rs [hunk 1]
```diff
@@ -213,22 +261,63 @@ impl Context {
...
         self.children = 0;
         self.parent_idx = Self::ROOT_INDEX;
-        self.simulation = is_simulation;
+        self.context_type = context_type;
         self.state = ContextState::Active;
     }
+
+    /// Destroy this context so it can no longer be used until it is re-initialized. The default
+    /// context cannot be re-initialized.
+    pub fn destroy(&mut self) {
```
[the remark this answers is not in the document]
Comment (zhalvorsen) [reply dpe/src/dpe_instance.rs:287]: I would think we could rely on the initialization, but I can add it if you think its necessary.

[the remark this answers is not in the document]
Comment (jhand2) [reply dpe/src/dpe_instance.rs:287]: That's fine, can leave this as-is

File: dpe/src/dpe_instance.rs [hunk 2]
```diff
@@ -213,22 +261,63 @@ impl Context {
...
+            // Once a default context is destroyed, it cannot be used until the next reset cycle.
+            ContextType::Default => ContextState::Locked,
+            ContextType::Simulation => ContextState::Inactive,
+        };
+    }
+}
+
+fn flags_iter(flags: u32, max: usize) -> FlagsIter {
+    FlagsIter {
+        flags: flags & !(u32::MAX << max),
```
[the remark this answers is not in the document]
Comment (zhalvorsen) [reply dpe/src/dpe_instance.rs]: Interesting that didn't come up in the unit tests, but good catch.

Done.

File: dpe/src/dpe_instance.rs [hunk 3]
```diff
@@ -213,22 +261,63 @@ impl Context {
...
+        self.context_type = context_type;
         self.state = ContextState::Active;
     }
+
+    /// Destroy this context so it can no longer be used until it is re-initialized. The default
+    /// context cannot be re-initialized.
+    pub fn destroy(&mut self) {
+        self.state = match self.context_type {
+            // Once a default context is destroyed, it cannot be used until the next reset cycle.
+            ContextType::Default => ContextState::Locked,
```
[the remark this answers is not in the document]
Comment (zhalvorsen) [reply dpe/src/dpe_instance.rs:294]: Done

File: dpe/src/commands.rs [hunk 4]
```diff
@@ -332,11 +393,21 @@ mod tests {
         }
     }
 
+    impl From<DestroyCtxCmd> for Vec<u8> {
+        fn from(value: DestroyCtxCmd) -> Self {
+            let mut raw = vec![];
+            raw.extend(value.handle);
+            raw.extend_from_slice(&value.flags.to_le_bytes());
+            raw
```
Comment (jhand2) [inline dpe/src/commands.rs:401]: Do you know how this works without dynamic allocation?

Comment (jhand2) [reply dpe/src/commands.rs:401]: oh jk, this is for tests

**Review by jhand2 — APPROVED** [review]

**Pull request merged**